src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/time.js.twig

```diff
@@ -0,0 +1,261 @@
+        return this.classnames[event.name] || this.classnames[event.category] || 'default';
+    }
+}
+
+class SvgRenderer {
+    /**
+     * @param  {SVGElement} element
+     */
+    constructor(element) {
+        this.ns = 'http://www.w3.org/2000/svg';
+        this.element = element;
+        this.defs = document.createElementNS(this.ns, 'defs');
+
+        this.add(this.defs);
+    }
+
+    get width() {
+        return this.element.clientWidth;
+    }
+
+    setViewBox(x, y, width, height) {
+        this.element.setAttribute('viewBox', `${x} ${y} ${width} ${height}`);
+    }
+
+    add(...elements) {
+        elements.forEach(element => this.element.appendChild(element));
+    }
+
+    addDef(...elements) {
+        elements.forEach(element => this.defs.appendChild(element));
+    }
+
+    group(...elements) {
+        const group = this.create('g');
+
+        elements.forEach(element => group.appendChild(element));
+
+        this.add(group);
+
+        return group;
+    }
+
+    createHorizontalLine(from, to, y, width, className) {
+        return this.set(
+            this.create('line'),
+            {
+                x1: from,
+                y1: y,
+                x2: to,
+                y2: y,
+                'stroke-width': width,
+                class: className,
+            }
+        );
+    }
+
+    createVerticalLine(from, to, x, width, className) {
+        return this.set(
+            this.create('line'),
+            {
+                x1: x,
+                y1: from,
+                x2: x,
+                y2: to,
+                'stroke-width': width,
+                class: className,
+            }
+        );
+    }
+
+    createFullHorizontalLine(y, width, className = 'border') {
+        const [from, , to] = this.element.getAttribute('viewBox').split(' ');
+
+        return this.createHorizontalLine(from, to, y, width, className);
+    }
+
+    createFullVerticalLine(x, width, className = 'border') {
+        const [, from, , to] = this.element.getAttribute('viewBox').split(' ');
+
+        return this.createVerticalLine(from, to, x, width, className);
+    }
+
+    createText(content, x, y, className, type = 'text') {
+        const text = this.set(
+            this.create(type),
+            { x, y, class: className }
+        );
+
+        text.textContent = content;
+
+        return text;
+    }
+
+    createBackground(border, className = 'background') {
+        const [x, y, width, height] = this.element.getAttribute('viewBox').split(' ');
+
+        return this.set(
+            this.create('rect'),
+            { x, y, width, height, 'stroke-width': border, class: className }
+        );
+    }
+
+    createPath(content) {
+        return this.set(this.create('path'), { d: content });
+    }
+
+    createMarker(id, refX, refY, markerWidth, markerHeight, orient) {
+        return this.set(
+            this.create('marker'),
+            { id, refX, refY, markerWidth, markerHeight, orient }
```

Comment:
using an id for the marker is a bad idea. When you have subrequests, you will have multiple timelines rendered on the page, and you don't make anything to keep ids unique here.
You should use a class for the CSS styles setting the color.

Reply:
It seems marker in SVG rely on CSS URI, so they _have_ to have ids to [be used](https://github.com/symfony/symfony/pull/30450/files#diff-cb59c7185481ff11f4e3645086b4664fR56)) (see [documentation](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/marker)).

Solutions:
- A: Don't use markers, draw standard shapes with SVG where they need to be.
- B: Keep the ids but define the marker globally in a global SVG file in the same page, and let both main request and sub-request graphs use them.
- C: there is actually a way to target them through class name that I might have missed?

I'm lean toward _solution A_.
Because markers can't be styled by their parents: I had to generate the same markers in different colors so far. So even if they are an elegant solution, their not optimal for our use case here.

What do you think?
Thanks for your valuable review.

Reply:
what are these markers used for in the visual representation ?

Reply:
The markers are used to draw the specific starting/ending shapes on period lines:
![capture d ecran 2019-03-06 a 09 17 52](https://user-images.githubusercontent.com/1846873/53866032-da60f900-3ff0-11e9-846f-a7bdc426bf6e.png) ![capture d ecran 2019-03-06 a 09 17 57](https://user-images.githubusercontent.com/1846873/53866035-dcc35300-3ff0-11e9-98e9-479b7704f648.png)

Reply:
then, you would also lean towards option A

Reply:
Done!